Doc/whatsnew/3.10.rst

```diff
@@ -659,10 +659,10 @@ are added to enable the warning.
 See :ref:`io-text-encoding` for more information.
 
 
-New Features Related to Type Annotations
```

Comment:
Personally, I'd rather use "annotations" here if you plan to add your entry under this section. ``Class and module objects now lazy-create empty annotations dicts on demand`` seems something related to annotations, not just type hinting ;). Or maybe we can just write both: ``New Features Related to Type Annotations and Hints``. What do you think?